### ISSUE TITLE
How do I get the package to work?

### ISSUE THREAD

**Issue by dentrinh** [osv/ac-html-csswatcher#3]
(defun setup-ac-for-html ()
  ;; Require ac-html since we are setup html auto completion
  (require 'ac-html)
  ;; Require default data provider if you want to use
  (require 'ac-html-default-data-provider)
  ;; Enable data providers,
  ;; currently only default data provider available
  (ac-html-enable-data-provider 'ac-html-default-data-provider)
  ;; Let ac-html do some setup
  (ac-html-setup)
  ;; Set your ac-source
  (setq ac-sources '(ac-source-html-tag
                     ac-source-html-attr
                     ac-source-html-attrv))

  (require 'ac-html-csswatcher)
  (setq ac-html-csswatcher-command "/usr/local/bin/csswatcher")
  (ac-html-csswatcher-setup)

  ;; Enable auto complete mode
  (auto-complete-mode))
(add-hook 'html-mode-hook 'setup-ac-for-html)

Nothing shows up in auto-completion-mode.
Can you elaborate a little more on the setup instructions in the READ.MD?

Thank you.

**Comment by osv** [comment]
Yes, I think I need create clean env and create test emacs config.

But first, can you help me to dig your problem?
1. Do you install csswatcher (`cpan install CSS::Watcher`)?
2. Can you show me next output of command?

```
csswatcher --help
```

And sorry for delay.

**Comment by dentrinh** [comment]
No worries.  Thanks for taking the time to respond.

1) Yes I installed csswatcher through CPAN and another dependency it required.  It was something File, I forgot what it was.

2) csswatcher --help outputs the following:

Smartmatch is experimental at /Library/Perl/5.18/CSS/Watcher.pm line 76.
Smartmatch is experimental at /Library/Perl/5.18/CSS/Watcher.pm line 148.

## csswatcher v0.4.2

Monitor project and create completion for ac-html

  Usage:
    /usr/local/bin/csswatcher [--help]
    /usr/local/bin/csswatcher [options] OBJECT

```
OBJECT......Directory or file, this script will find project home like projectile
socket......Socket file, default
            ~/.emacs.d/ac-html-csswatcher/csswatcher.socket
version.....Print version
```

Daemon options:
    daemon......Start daemon (it automatically start if OBJECT defined).
    restart.....Restart daemon
    stop........Stop daemon
    logfile.....Set log file for daemon, default
                ~/.emacs.d/ac-html-csswatcher/csswatcher.log
    outputdir...Output root directory where place generated files for ac-html
                ~/.emacs.d/ac-html-csswatcher/completion/
    pidfile.....PID file for daemon, default
                ~/.emacs.d/ac-html-csswatcher/csswatcher.pid
    debug.......Debug level, default INFO. You can set
                OFF FATAL ERROR WARN INFO DEBUG TRACE ALL.

See also https://github.com/cheunghy/ac-html

**Comment by osv** [comment]
Looks you install it :)

I'm little busy now for testing, later I will check in clean instalation, but I cant do it now, need 1-2 week, sorry!

**Comment by osv** [comment]
Confirmed, there is some problems... I fix it soon

**Comment by dentrinh** [comment]
Awesome.  Looking forward to check it out.

**Comment by osv** [comment]
Hello, you can update it from melpa. If you have problem to make it work let me know

**Comment by dentrinh** [comment]
Sorry but the update did not work unfortunately.

**Comment by osv** [comment]
Ok, lets dig  deeper then :)

You may need run `M-x ac-html-csswatcher+` and `M-x ac-html-csswatcher-refresh` when editing html file...

Tell me please

Are you use `company-mode` or `ac-html`? I just install ac-html and suddenly it not work :)

**Comment by dentrinh** [comment]
<blockquote>You may need run M-x ac-html-csswatcher+ and M-x ac-html-csswatcher-refresh when editing html file...</blockquote>

Yeah I did this.  

I am using auto-complete with ac-html as posted in the first comment and that is not working.

**Comment by osv** [comment]
Try install stable version of ac-html v0.31 it looks recent version ac-html incompatible with ac-html-csswatcher and ac-html-bootstrap for now :(
Or you can try to use `company`..

**Comment by osv** [comment]
If you need help with company I can help, but with auto-complete I need more time to dig

**Comment by dentrinh** [comment]
Thanks for all the help.  I will give these suggested solutions a go later today.

**Comment by osv** [comment]
Let me know if you need help

**Comment by dentrinh** [comment]
Thanks for all your quick responses.  I installed the stable version and I don't see css completions, though I get completions back again.  Unfortunately I will tinker with this some more later because I am busy for the next week.

**Comment by dentrinh** [comment]
Finally got some time to come back to this and I got it to work now.  I hooked company-web-csswatcher+ and opted to use company-web for the auto-completion because that works correctly.  I had to add a .csswatcher and ignored all .css/.less/.sass files at my project root because it was scanning third party packages and getting parsing errors.  I then made exceptions on which stylesheets to parse and it worked like a charm.  Thanks again for another awesome package.
